R/make_quantile_reg.R

```diff
@@ -102,9 +102,9 @@ make_quantile_reg <- function() {
     # can't make a method because object is second
     out <- switch(
       type,
-      rq = dist_quantiles(x, object$tau), # one quantile
+      rq = dist_quantiles(unname(as.list(x)), object$tau), # one quantile
       rqs = {
-        x <- unname(apply(x, 1, function(q) unname(sort(q)), simplify = FALSE))
+        x <- unname(as.list(as.data.frame(apply(x, 1, function(q) unname(sort(q))))))
```

Comment:
```r
x <- unname(split(apply(x, 1, function(q) sort(q)), seq(nrow(x)))
```

Reply:
Yes. I think we could simplify it… though when I replace the code with that line & run the quantile_reg() doc. example, I get this error:

```
Error:
! `q[order(tau)]` produces unsorted quantiles.
Run `rlang::last_trace()` to see where the error occurred.
Called from: signal_abort(cnd, .file)
```
I think that a slight change of order of operations should do the trick... Let’s run through an ex based on the documentation & let me know if it does what you think it should.

Suppose we’re working with this temp (x just before going through that bit of code), which is a 100 x 2 matrix:
```
head(xtemp)
    tau= 0.2   tau= 0.8
1  0.4681302 -1.2231489
2 -1.2828086  0.6921821
3  0.4660981 -0.9436714
4 -1.3417686  0.6614161
5 -1.0832523  0.5037221
6 -1.2014149  0.4241279
```

So, 1st split the df into a list as you said:
```
> head(split(xtemp, seq(nrow(xtemp))))
$`1`
[1]  0.4681302 -1.2231489

$`2`
[1] -1.2828086  0.6921821

$`3`
[1]  0.4660981 -0.9436714

$`4`
[1] -1.3417686  0.6614161

$`5`
[1] -1.0832523  0.5037221

$`6`
[1] -1.2014149  0.4241279
```
Then unname the list:
```
> unname(head(split(xtemp, seq(nrow(xtemp)))))
[[1]]
[1]  0.4681302 -1.2231489

[[2]]
[1] -1.2828086  0.6921821

[[3]]
[1]  0.4660981 -0.9436714

[[4]]
[1] -1.3417686  0.6614161

[[5]]
[1] -1.0832523  0.5037221

[[6]]
[1] -1.2014149  0.4241279
```
And use lapply to sort each pair.
```
> head(lapply(unname(split(xtemp, seq(nrow(xtemp)))), function(q) sort(q)))
[[1]]
[1] -1.2231489  0.4681302

[[2]]
[1] -1.2828086  0.6921821

[[3]]
[1] -0.9436714  0.4660981

[[4]]
[1] -1.3417686  0.6614161

[[5]]
[1] -1.0832523  0.5037221

[[6]]
[1] -1.2014149  0.4241279
```
So all together we get that `x <- lapply(unname(split(x, seq(nrow(x)))), function(q) sort(q))`